clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -1681,7 +1681,7 @@ std::optional<unsigned> Sema::GetDecompositionElementCount(QualType T,
   llvm::APSInt TupleSize(Ctx.getTypeSize(Ctx.getSizeType()));
   switch (isTupleLike(*this, Loc, T, TupleSize)) {
   case IsTupleLike::Error:
-    return {};
+    return std::nullopt;
```

Comment:
how are those changes related to the other one / the commit message?

Reply:
Just drive by cleanup

Reply:
doesn't the change on 1709 change the return value?

Reply:
Yes, it should not have, as was pointed out in the original PR (and it's clearly bogus).
I edited the commit message to clarify.
Thanks!